src/prefect/task_engine.py

```diff
@@ -506,6 +516,18 @@ def record_terminal_state_timing(self, state: State) -> None:
                         state.timestamp - self.task_run.state.timestamp
                     )
 
+    def handle_rollback(self, txn: Transaction) -> None:
+        # transaction rollbacks can occur outside of the engine's context
+        # so we need to ensure a client is available
+        with ClientContext.get_or_create() as client_ctx:
```

Comment:
With client-side task orchestration, we don't need this, but making it work with or w/o keeps the test suite happier.